test/models/host_test.rb

```diff
@@ -1557,6 +1557,15 @@ def teardown
       assert_equal '10.10.0.1', host.interfaces.where(:identifier => 'eth0_0').first.ip
     end
 
+    test "#set_interfaces updates existing bridge interface suggested as primary" do
+      host, parser = setup_host_with_nic_parser({:identifier => 'br-ex', :macaddress => '00:00:00:11:22:33', :bridge => true, :virtual => true, :ipaddress => '10.10.0.1'})
+      host.managed = false
+      host.primary_interface.update(:identifier => 'br-ex', :mac => '00:00:00:11:22:33', :ip => '10.10.0.1')
+
+      host.set_interfaces(parser)
+      assert host.interfaces.where(:identifier => 'br-ex').first.virtual
+    end
```

Comment:
Was the unset virtual flag after update the root problem?

Reply:
some kind of